pandas/core/arrays/datetimes.py

```diff
@@ -1756,9 +1764,13 @@ def _generate_range_overflow_safe(endpoint, periods, stride, side='start'):
     Parameters
     ----------
     endpoint : int
```

Comment:
I would like you to move this out of this file; you can make a subdirectory ``pandas/core/arrays/datetimes`` if you want or make this a private module

Reply:
OK.  Maybe `arrays._ranges`?  Also opens a natural place if we ever want to have a RangeArray.

could also move timedeltas._generate_regular_range there (and looking at it, it seems plausible it would have the same bug as the datetime version.